### PR TITLE
Shape embed

### PR DESCRIPTION
**Pull request by ctr26**
- Adding standard scalar to the features->scoring model
- Putting umap back in but as a function
- [Check] Seeding the analysis for running repeats
- k_folds 5 as 10 might be to many slices
- update args to match what experiments found was good
- [Check], attempted to implement proper best_weight finding 
  - It should, run epochs until end if interrupted
  - When it does run epochs to end, load in best weights (val loss minimum)

PR failing because of disk space

### CODE REVIEW

File: scripts/shapes/shape_embed.py [hunk 1]
```diff
@@ -57,12 +62,54 @@
...
+        ax,
+        "upper center",
+    )
+    ax.set(xlabel=None, ylabel=None)
+    sns.despine(left=True, bottom=True)
+    plt.tick_params(bottom=False, left=False, labelbottom=False, labelleft=False)
+    plt.tight_layout()
+    plt.savefig(metadata(f"umap_no_axes.pdf"))
+    # plt.show()
+    plt.close()
```
Comment (afoix) [inline scripts/shapes/shape_embed.py:110]: We agree to put back the UMAP to see how does it look now 😄

File: scripts/shapes/shape_embed.py [hunk 2]
```diff
@@ -77,14 +124,20 @@ def scoring_df(X, y):
...
 
     # Create a random forest classifier
-    clf = RandomForestClassifier()
+    pipeline = Pipeline(
+        [
+            ("scaler", StandardScaler()),
+            #  ("pca", PCA(n_components=0.95, whiten=True, random_state=42)),
+            ("clf", RandomForestClassifier()),
+        ]
+    )
```
Comment (afoix) [inline scripts/shapes/shape_embed.py:133]: Use scalar previous `RandomForestClassifier`

File: scripts/shapes/shape_embed.py [hunk 3]
```diff
@@ -77,14 +124,20 @@ def scoring_df(X, y):
...
 
     # Specify the number of folds
-    k_folds = 10
+    k_folds = 5
```
Comment (afoix) [inline scripts/shapes/shape_embed.py:136]: we agree to change and see from 10 to 5 (datasets are not that big)

Comment (afoix) [reply scripts/shapes/shape_embed.py:136]: Changing from 10 to 5 kfolds because with our sizes of datasets 5should be enough.

File: scripts/shapes/shape_embed.py [hunk 4]
```diff
@@ -115,26 +168,18 @@ def shape_embed_process():
     window_size = 128 * 2
 
     params = {
-        "model":"resnet18_vqvae_legacy",
-        "epochs": 75,
+        "model": "resnet50_vqvae",
+        "epochs": 250,
```
Comment (afoix) [inline scripts/shapes/shape_embed.py:172]: Increment epochs to be sure that larger models flatten.

File: scripts/shapes/shape_embed.py [hunk 5]
```diff
@@ -115,26 +168,18 @@ def shape_embed_process():
...
-        # "embedding_dim": 32,
-        # "num_embeddings": 16,
-        "commitment_cost": 0.25,
-        "decay": 0.99,
         "frobenius_norm": False,
     }
 
     optimizer_params = {
-        "opt": "LAMB",
+        "opt": "AdamW",
```
Comment (afoix) [inline scripts/shapes/shape_embed.py:182]: Change optimizer

File: scripts/shapes/shape_embed.py [hunk 6]
```diff
@@ -306,11 +351,17 @@ def shape_embed_process():
...
     Path(f"{model_dir}/").mkdir(parents=True, exist_ok=True)
 
-    checkpoint_callback = ModelCheckpoint(dirpath=f"{model_dir}/", save_last=True)
+    checkpoint_callback = ModelCheckpoint(
+        dirpath=f"{model_dir}/",
+        save_last=True,
+        save_top_k=1,
+        monitor="loss/val",
+        mode="min",
+    )
```
Comment (afoix) [inline scripts/shapes/shape_embed.py:360]: Making sure than what it shows as is the best model monitoring the validation loss.

**Review by afoix — APPROVED** [review]
@ctr26 I just make comments to me to understand each of the bits and to see if I was following everything but I am happy with every single bit you modify so go ahead and merge if you like to do so 🥳
